allennlp/modules/default_lstm.py

```diff
@@ -0,0 +1,36 @@
+import re
+from overrides import overrides
+import torch
+
+from allennlp.nn.initializers import block_orthogonal
+
+
+class DefaultLstm(torch.nn.LSTM):
+    """
+    A wrapper of the Pytorch LSTM which provides a good default initialization
+    of weights and biases, and disables training of the superfluous bias_hh parameters.
+
+    Input parameters are identical to Pytorch LSTM.
+    """
+    @overrides
+    def reset_parameters(self):
+        hidden_size = self.hidden_size
+        for param_name, param in self.named_parameters():
+            if re.search("bias_hh", param_name):
```

Comment:
You shouldn't have to do a search here; you should get the name exactly, and can just match on that (`if name == "biah_hh":`).  Because you're calling `self.named_parameters()`, there's no nesting of names, or anything - you'll just get `"bias_hh"`, etc., as the names.

Reply:
In fact, you can just remove the whole loop over `self.named_parameters()`, and just directly initialize the things you want to initialize.  It's a tiny bit less robust to future changes in pytorch, but I think that's a feature, not a bug - if pytorch changes their LSTMs enough to break this, we will want to do things differently, too.

Reply:
There are the different layers and directions though, e.g., here's a full list for a 2-layer bidirectional one: `['weight_hh_l1_reverse', 'bias_hh_l1', 'bias_ih_l0', 'weight_ih_l1', 'weight_hh_l0', 'bias_ih_l1_reverse', 'bias_ih_l0_reverse', 'bias_hh_l1_reverse', 'weight_ih_l0', 'bias_hh_l0', 'weight_ih_l1_reverse', 'bias_ih_l1', 'weight_ih_l0_reverse', 'weight_hh_l0_reverse', 'weight_hh_l1', 'bias_hh_l0_reverse']`. I can generate that list, assuming nothing changes (and as you say, if it does, it might be a feature to find out), but this seemed cleaner.

Reply:
Oh, yep, sorry, my bad.  You're right.

Reply:
nit: it looks like in every case you're really just checking `param_name.startswith()`? if so, I feel like it's more readable that way.

Reply:
Sure, I'll change that